Homework3/task_2.py

```diff
@@ -0,0 +1,230 @@
+import os
+import logging
+import csv
+import requests
+import argparse
+import shutil
+from datetime import datetime, timedelta
+from statistics import mean
+from collections import Counter
+
+URL = 'https://randomuser.me/api/?results=5000&format=csv'
+
+
+def setup_logger():
+    logging.basicConfig(
+        filename='my_log_file.log',
+        level=logging.DEBUG,  #
+        format='%(asctime)s - %(levelname)s - %(message)s"'
+    )
+    return logging.getLogger('my_logger')
+
+
+my_logger = setup_logger()
+
+
+def download_user_accounts_csv(output_csv):
+    my_logger.info('Downloading user accounts data from the API...')
+    response = requests.get(URL)
+    if response.status_code == 200:
+        data = response.text
+        if data:
+            my_logger.info(f'Received user accounts from the API.')
+            with open(output_csv, 'w', newline='', encoding='utf-8') as file:
+                file.write(data)
+            my_logger.info('Successfully downloaded user accounts to user_accounts.csv')
+        else:
+            my_logger.warning('No data received from the API.')
+    else:
+        my_logger.error(f'Failed to fetch data. Status code: {response.status_code}')
+
+
+def read_data_from_csv():
+    with open('user_accounts.csv', 'r', newline='', encoding='utf-8') as csvfile:
+        reader = csv.DictReader(csvfile)
+        data = [row for row in reader]
+    return data
+
+
+def write_data_in_csv(file_path, data1):
+    with open(file_path, 'w', newline='', encoding='utf-8') as csvfile:
+        fieldnames = list(data1[0].keys())
+        writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+        writer.writeheader()
+        writer.writerows(data1)
```

Comment:
**data1** -- is  a bad naming. I suspect you used **1** to avoid shadowing of another var from global scope. but it's a wrong decision.